### PR TITLE
WIP Add types to every class that extends Items

### PR DESCRIPTION
**Pull request by apstanisic**
I wanted to try to bring full types support to all handlers. This is the easiest, simplest and cleanest way to do it. I had to change generics in `ItemsHandler`. Example of new usage is in `SettingsHandler`. I first wanted to see if you are okay with api changes before adding types to all handlers.
Also, i think that `Response.data` never returns null. If user does not have permission or item does not exist, it will throw error. If fetching array, it will return empty array.

What do you think?

### CODE REVIEW

File: packages/sdk-js/src/handlers/activity.ts [hunk 1]
```diff
@@ -1,24 +1,43 @@
...
+	action: string;
+	ip: string;
+	item: PrimaryKey;
+	user_agent: string;
+	timestamp: string;
+	id: number;
+	user: string;
+	comment: string | null;
+	collection: string;
+	revisions: [number] | null;
```
Comment (apstanisic) [inline packages/sdk-js/src/handlers/activity.ts:15]: When I tested the api, `revisions` always return either null or array with single number inside. Maybe I don't have enough activity data to get different results

**Comment by rijkvanzanten** [comment]
The problem with the types of those is that all fields should be considered optional (as the access is permissions based, and is different per user), and that every type can be extended with custom fields, as directus_* collections can be extended with your own fields 🤔

**Comment by apstanisic** [comment]
I've added support for `SettingsHandler` to extend with custom fields, but it's impossible to add types to getter in index.ts file.
And I've set by default all returned value to be partial if type is specified, with support for overwriting per method
Screenshot from 2020-11-25 07-17-52

**Comment by apstanisic** [comment]
See #4510

**Pull request closed without merging**